dali/base/dasds.cpp

```diff
@@ -1377,56 +1385,67 @@ class CDeltaWriter : implements IThreaded
             threaded.join();
         }
     }
+    bool isAborted() const { return aborted; }
 // IThreaded
     virtual void threadmain() override
     {
-        while (!aborted)
+        PROGLOG("CDeltaWriter thread started");
+        try
         {
-            bool semTimedout = false;
-            if (saveThresholdSecs)
-                semTimedout = !pendingTransactionsSem.wait(saveThresholdSecs * 1000);
-            else
-                pendingTransactionsSem.wait();
-
-            if (aborted)
-                break;
-            // keep going whilst there's things pending
-            while (true)
+            while (!aborted)
             {
-                CLeavableCriticalBlock b(pendingCrit);
-                std::queue<Owned<CTransactionItem>> todo = std::move(pending);
-                if (0 == todo.size())
+                bool semTimedout = false;
+                if (saveThresholdSecs)
+                    semTimedout = !pendingTransactionsSem.wait(saveThresholdSecs * 1000);
+                else
+                    pendingTransactionsSem.wait();
+
+                if (aborted)
+                    break;
+                // keep going whilst there's things pending
+                while (true)
                 {
-                    if (writeRequested)
+                    CLeavableCriticalBlock b(pendingCrit);
+                    std::queue<Owned<CTransactionItem>> todo = std::move(pending);
+                    if (0 == todo.size())
                     {
-                        // NB: if here, implies someone signalled via requestAsyncWrite()
+                        if (writeRequested)
+                        {
+                            // NB: if here, implies someone signalled via requestAsyncWrite()
 
-                        // if reason we're here is because sem timedout, consume the signal that was sent
-                        if (semTimedout)
-                            pendingTransactionsSem.wait();
+                            // if reason we're here is because sem timedout, consume the signal that was sent
+                            if (semTimedout)
+                                pendingTransactionsSem.wait();
 
-                        writeRequested = false;
-                    }
-                    if (signalWhenAllWritten)
-                    {
-                        signalWhenAllWritten = false;
-                        allWrittenSem.signal();
+                            writeRequested = false;
+                            if (signalWhenAllWritten) // can only be true if writeRequested was true
```

Comment:
@ghalliday: NB: I moved this inside the if (writeRequested) block, so it is clearer that is only sent and possible when writeRequested is true